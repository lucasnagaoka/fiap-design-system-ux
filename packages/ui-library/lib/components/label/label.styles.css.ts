import { style } from '@vanilla-extract/css';

import { FIAP_DS_COLORS } from '../../design-tokens/colors';
import { FIAP_DS_SPACING } from '../../design-tokens/spacing';
import { FIAP_DS_TYPOGRAPHY } from '../../design-tokens/typography';

const label = style({
  padding: FIAP_DS_SPACING.s2,
  width: '100%',
  fontSize: FIAP_DS_TYPOGRAPHY.small['font-size'],
  lineHeight: FIAP_DS_TYPOGRAPHY.small['line-height'],
  fontWeight: FIAP_DS_TYPOGRAPHY.small.weight,
  backgroundColor: FIAP_DS_COLORS.white[100],
  color: FIAP_DS_COLORS.primary[500],
});

export { label };
