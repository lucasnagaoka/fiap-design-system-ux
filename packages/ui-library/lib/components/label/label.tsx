import clsx from 'clsx';
import { fontText } from '../../styles/font.css';
import { label } from './label.styles.css';

import type { JSX } from 'react';
import type { LabelProps } from './label.types';

export function Label({ children, ...other }: LabelProps): JSX.Element {
  return (
    <label className={clsx([fontText, label])} {...other}>
      {children}
    </label>
  );
}

Label.displayName = 'Label';
