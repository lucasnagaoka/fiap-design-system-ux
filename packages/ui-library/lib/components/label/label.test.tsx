import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Label } from './label';

describe('Label component', () => {
  it('Should render a Label', () => {
    render(<Label htmlFor='label' children='Label' />);

    expect(screen.getByText('Label')).toBeTruthy();
  });
});
