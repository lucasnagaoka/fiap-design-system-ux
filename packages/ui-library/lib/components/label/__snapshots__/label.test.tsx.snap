// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label component > Should take a label snapshot 1`] = `[Function]`;
